### ISSUE TITLE
panic: Got string in parse_basic_expression (0.8.0 regression)

### ISSUE THREAD

**Issue by southerntofu** [getzola/zola#748]
# Bug Report

## Environment

Zola version: 0.8.0

## Expected Behavior

The site should build.

## Current Behavior

Website build panics.

```
$ RUST_BACKTRACE=1 zola build
Building site...
thread 'main' panicked at 'internal error: entered unreachable code: Got string in parse_basic_expression', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/parser/mod.rs:268:14
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:381
   6: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:336
   7: tera::parser::parse_basic_expression
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/parser/mod.rs:268
   8: tera::parser::parse_macro_fn
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/parser/mod.rs:690
   9: tera::parser::parse_macro_definition
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/parser/mod.rs:717
  10: tera::parser::parse_content
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/parser/mod.rs:934
  11: tera::parser::parse
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/parser/mod.rs:1091
  12: tera::template::Template::new
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/template.rs:48
  13: tera::tera::Tera::add_file
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/tera.rs:183
  14: tera::tera::Tera::load_from_glob
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/tera.rs:149
  15: tera::tera::Tera::create
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/tera.rs:62
  16: tera::tera::Tera::parse
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.0.0-beta.10/src/tera.rs:105
  17: site::Site::new
             at /home/user/Prog/zola/components/site/src/lib.rs:97
  18: zola::cmd::build::build
             at src/cmd/build.rs:9
  19: zola::main
             at src/main.rs:50
  20: std::rt::lang_start::{{closure}}
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/libstd/rt.rs:64
  21: std::panicking::try::do_call
             at src/libstd/rt.rs:49
             at src/libstd/panicking.rs:293
  22: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:87
  23: std::rt::lang_start_internal
             at src/libstd/panicking.rs:272
             at src/libstd/panic.rs:388
             at src/libstd/rt.rs:48
  24: std::rt::lang_start
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/libstd/rt.rs:64
  25: main
  26: __libc_start_main
  27: _start
```

The website builds on previous versions (0.6.0, 0.7.0). Thanks for making Zola awesome <3

**Comment by Keats** [comment]
Do you have the templates causing that somewhere by any chance?

**Comment by southerntofu** [comment]
I just published it [here](https://tildegit.org/southerntofu/staticadventures), thanks

**Comment by Keats** [comment]
This is fixed in Tera beta 13 and will be in 0.8.1, you can use the `next` branch in the meantime

**Comment by southerntofu** [comment]
Thank you, it works like a charm!
